File: Year II/First Semester/MAP/Laborator/Laborator2/model/QueueContainer.java
package sem1_2.model;

public class QueueContainer extends AbstractTaskContainer {
    @Override
    public Task remove(){
        if(!tasks.isEmpty())
            return tasks.remove(0);
        return null;
    }
}

/*
before extracting superclass

public class QueueContainer implements Container{
    private final List<Task> tasks = new ArrayList<Task>();

    @Override
    public Task remove() {
        if(!tasks.isEmpty())
            return tasks.remove(0);
        else return null;
    }

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}
*/

File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Model/QueueContainer.cs
namespace ConsoleApp1.Model;

public class QueueContainer: AbstractTaskContainer
{
    public override Task Remove()
    {
        if(Tasks.Count == 0)
            throw new InvalidOperationException("Stack container is empty");
        
        Task removedTask = Tasks[0];
        Tasks.RemoveAt(0);
        
        return removedTask;
    }
}